ui/lib/kubernetes/addon/components/page/role/create-and-edit.hbs

```diff
@@ -0,0 +1,139 @@
+  </h2>
+  <hr class="is-marginless has-background-gray-200" />
+</div>
+
+{{#if @model.generationPreference}}
+  <form id="role" {{on "submit" this.onSave}} data-test-policy-form>
+    {{#each @model.filteredFormFields as |field|}}
+      <FormField @attr={{field}} @model={{@model}} @modelValidations={{this.modelValidations}} />
+    {{/each}}
+
+    <div class="has-bottom-margin-s">
+      <ToggleButton
+        data-test-field="annotations"
+        @isOpen={{this.showAnnotations}}
+        @openLabel="Hide annotations and labels"
+        @closedLabel="Annotations and labels"
+        @onClick={{fn (mut this.showAnnotations)}}
+      />
+      {{#if this.showAnnotations}}
+        <div class="box" data-test-annotations>
+          {{#each this.extraFields as |field|}}
+            <div class={{if (eq field.type "labels") "has-top-margin-xl"}}>
+              <h2 class="title is-4">Extra {{field.type}}</h2>
+              <p>
+                {{field.description}}
+                See
+                <ExternalLink @href="https://kubernetes.io/docs/concepts/overview/working-with-objects/{{field.type}}/">
+                  Kubernetes
+                  {{singularize field.type}}
+                  documentation here
+                </ExternalLink>.
+              </p>
+              <KvObjectEditor
+                class="has-top-margin-m"
+                data-test-kv={{field.type}}
+                @value={{get @model field.key}}
+                @onChange={{fn (mut (get @model field.key))}}
+              />
+            </div>
+            <hr />
+          {{/each}}
+        </div>
+      {{/if}}
+    </div>
+
+    {{#if (eq @model.generationPreference "full")}}
+      <div class="has-top-margin-m has-bottom-margin-l" data-test-generated-role-rules>
+        <h2 class="title is-4">
+          Generated role rules
+        </h2>
+        <FormFieldLabel
+          for="templates"
+          @label="Role rules template"
+          @subText="Start with a template for role rules based on your use case"
+        />
+        <div class="select is-fullwidth">
+          <select id="templates" data-test-select-template {{on "change" this.selectTemplate}}>
+            {{#each this.roleRulesTemplates as |template|}}
+              <option selected={{eq this.selectedTemplateId template.id}} value={{template.id}}>
+                {{template.label}}
+              </option>
+            {{/each}}
+          </select>
+        </div>
+        {{#let (find-by "id" this.selectedTemplateId this.roleRulesTemplates) as |template|}}
+          <JsonEditor
+            class="has-top-margin-l"
+            data-test-rules
+            @title="Role rules"
+            @value={{template.rules}}
+            @mode="ruby"
+            @valueUpdated={{fn (mut template.rules)}}
+            @helpText={{this.roleRulesHelpText}}
+          >
+            <button type="button" class="toolbar-link" onClick={{this.resetRoleRules}} data-test-restore-example>
```

Comment:
I'm not sure if this makes a difference, but I noticed that Octane mostly uses `{{on "click" this.method}}` rather than `onClick` for click actions

Reply:
Yes good catch! I copied that from somewhere else and missed that. I will update.